webrtc.go

```diff
@@ -105,17 +108,51 @@ func newSession(config SessionConfig) (*Session, error) {
 		scopedLogger.Warn().Err(err).Msg("Failed to create PeerConnection")
 		return nil, err
 	}
+
 	session := &Session{peerConnection: peerConnection}
 	session.rpcQueue = make(chan webrtc.DataChannelMessage, 256)
+	session.hidQueue = make(chan webrtc.DataChannelMessage, 1024)
+
 	go func() {
 		for msg := range session.rpcQueue {
-			onRPCMessage(msg, session)
+			// TODO: only use goroutine if the task is asynchronous
+			go onRPCMessage(msg, session)
+		}
+	}()
+
+	go func() {
+		for msg := range session.hidQueue {
+			onHidMessage(msg.Data, session)
 		}
 	}()
 
 	peerConnection.OnDataChannel(func(d *webrtc.DataChannel) {
+		defer func() {
+			if r := recover(); r != nil {
+				scopedLogger.Warn().Interface("error", r).Msg("Recovered from panic in DataChannel handler")
```

Comment:
```golang
scopedLogger.Error().Err(r),Msg("Recovered from panic in DataChannel handler")
```

 so it's an error-level log, and has the error details formatted as an Error

Reply:
I think we still want to use `.Err(r)` instead of `.Interface("Error", r)` here.

Reply:
I felt the same until I found that the return type of `recover()` is `interface{}`, rather than `Error`. Are you surprised too? :-)

Reply:
Not really surprised, since as far as I can tell `Interface {}` is just `any` 😄

The difference is I was going for is `Err(` is (to the logger) known to be an error object, so the logging doesn't need reflection to log out the details as `Interface(` does.

Either is fine, as hopefully we never hit this code anyway 😁